server/fishtest/rundb.py

```diff
@@ -76,6 +76,46 @@ def __init__(self, db_name="fishtest_new", port=-1, is_primary_instance=True):
 
         self.__is_primary_instance = is_primary_instance
 
+        self.task_lock = threading.Lock()
+        self.request_task_lock = threading.Lock()
+        self.timer = None
+        self.timer_active = True
+
+    def set_inactive_run(self, run):
+        with self.task_lock:
+            run["workers"] = run["cores"] = 0
+            for task in run["tasks"]:
+                task["active"] = False
+
+    def set_inactive(self, task_id, run):
```

Comment:
What about `def set_inactive_task(self, task_id, run):` ?

Reply:
Good idea! I changed it.